samza-kafka/src/main/scala/org/apache/samza/system/kafka/KafkaSystemAdmin.scala

```diff
@@ -542,6 +562,20 @@ class KafkaSystemAdmin(
   }
 
   /**
+    * @inheritdoc
+    *
+    * Delete records of partitions in Kafka up to (and include) the corresponding offsets. This only works with Kafka cluster 0.11 or later.
+    * Otherwise it's a no-op.
+    */
+  override def deleteRecords(offsets: util.Map[SystemStreamPartition, String]) {
+    if (adminClient == null)
+      throw new SamzaException("KafkaSystemAdmin has not started yet")
+    adminClient.deleteRecordsBefore(offsets.asScala.toSeq.map { case (systemStreamPartition, offset) =>
```

Comment:
Prefer extracting variable for the argument to deleteRecordsBefore.

Reply:
Sure. It is fixed now.